token/program-2022/src/processor.rs

```diff
@@ -841,6 +841,11 @@ impl Processor {
             {
                 confidential_transfer_state.closable()?
             }
+
+            if let Ok(transfer_fee_state) = source_account.get_extension_mut::<TransferFeeAmount>()
```

Comment:
Mebbe use immutable `get_extension` when it lands? https://github.com/solana-labs/solana-program-library/pull/2822/files#diff-1fd54e31043386c786ab6a3edf57cd653cf69082500a805e9dd50f550d2f260bR443

Reply:
Heck yes, I'll add a TODO for that, thanks!